### PR TITLE
mon/OSDMonitor: Cleanup osdmap.get_num_up_osd, etc

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -3097,7 +3098,7 @@ void OSDMonitor::check_osdmap_sub(Subscription *sub)
 
 void OSDMonitor::check_pg_creates_subs()
 {
-  if (!osdmap.get_num_up_osds()) {
+  if (osdmap.get_num_up_osds() == 0) {
```
Comment (amitkumar50) [inline src/mon/OSDMonitor.cc]: Will it make any difference?
changing ! to ==

File: src/mon/OSDMonitor.cc [hunk 2]
```diff
@@ -10956,7 +10957,19 @@ bool OSDMonitor::prepare_pool_op_create(MonOpRequestRef op)
...
 				   ostream *ss)
 {
-  const string& poolstr = osdmap.get_pool_name(pool_id);
+  const std::string& poolstr = osdmap.get_pool_name(pool_id);
+
+  // Check permission
+  if (!g_conf->mon_allow_pool_delete) {
+    *ss << "pool deletion is disabled; you must first set the "
+        << "mon_allow_pool_delete config option to true before you can "
+        << "destroyi a pool";
```
Comment (amitkumar50) [inline src/mon/OSDMonitor.cc]: destroyi?

File: src/mon/OSDMonitor.cc (hunk 1 above)
Comment (shinobu-x) [reply src/mon/OSDMonitor.cc]: Should not make not any difference on its behavior. However we should use same expression as much as possible :( 

https://github.com/ceph/ceph/blob/b4e4ad3b855f86d2d6f7d07c670c79e3dab59655/src/mon/OSDMonitor.cc#L1369

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
@jecluis 
Would you please review the branch?

**Comment by liewegas** [comment]
Can we not add the 'else' between the ifs?  When the block returns the else is unnecessary.

File: src/mon/OSDMonitor.cc (hunk 2 above)
Comment (jecluis) [reply src/mon/OSDMonitor.cc]: What @amitkumar50 said. Also, please have `mon_allow_pool_delete` between apostrophes, so it is clearer what the config option name is, as well as `true`, as the value to be set to. Bonus points if you end the statement with a period. :)

File: src/mon/OSDMonitor.cc [hunk 3]
```diff
@@ -10956,7 +10957,19 @@ bool OSDMonitor::prepare_pool_op_create(MonOpRequestRef op)
...
+    return -EPERM;
+  } else if (pool.has_flag(pg_pool_t::FLAG_NODELETE)) {
+    *ss << "pool deletion is disabled; you must unset nodelete flag for the "
+        << "pool first";
```
Comment (jecluis) [inline src/mon/OSDMonitor.cc]: I would phrase it as `you must first unset the 'nodelete' flag for this pool.`

**Comment by tchaikov** [comment]
i am closing this PR due to lack of activity, please feel free to reopen it when you are back on it =)

**Pull request closed without merging**